osu.Game/Beatmaps/Formats/LegacyDecoder.cs

```diff
@@ -126,7 +128,9 @@ protected void HandleColours<TModel>(TModel output, string line, bool allowAlpha
             string[] split = pair.Value.Split(',');
             Color4 colour = convertSettingStringToColor4(split, allowAlpha, pair);
 
-            bool isCombo = pair.Key.StartsWith(@"Combo", StringComparison.Ordinal);
+            bool isCombo = pair.Key.StartsWith(@"Combo", StringComparison.Ordinal)
+                           && int.TryParse(pair.Key[5..], out int comboIndex)
+                           && comboIndex >= 1 && comboIndex <= MAX_COMBO_COLOUR_COUNT;
```

Comment:
By the way, I am choosing not to notice that none of this parsing logic actually *checks* that the combo colours are in order and just parses them sequentially without examining the number in the key, because it's very annoying to handle. Until someone invariably reports that as an issue, that is.